osde2e/managed_upgrade_operator_tests.go

```diff
@@ -0,0 +1,234 @@
+// DO NOT REMOVE TAGS BELOW. IF ANY NEW TEST FILES ARE CREATED UNDER /osde2e, PLEASE ADD THESE TAGS TO THEM IN ORDER TO BE EXCLUDED FROM UNIT TESTS.
+//go:build osde2e
+// +build osde2e
+
+package osde2etests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apiv1 "github.com/openshift/api/config/v1"
+	config "github.com/openshift/client-go/config/clientset/versioned"
+	upgradev1alpha1 "github.com/openshift/managed-upgrade-operator/api/v1alpha1"
+	"github.com/openshift/osde2e-common/pkg/clients/openshift"
+	"github.com/openshift/osde2e-common/pkg/clients/prometheus"
+	"github.com/openshift/osde2e-common/pkg/gomega/assertions"
+	. "github.com/openshift/osde2e-common/pkg/gomega/matchers"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	operatorName                           = "managed-upgrade-operator"
+	operatorNamespace                      = "openshift-managed-upgrade-operator"
+	operatorLockFile                       = "managed-upgrade-operator-lock"
+	upgradeConfigResourceName              = "managed-upgrade-config"
+	upgradeConfigForDedicatedAdminTestName = "osde2e-da-upgrade-config"
+	defaultDesiredReplicas                 = 1
+	rolePrefix                             = "managed-upgrade-operator"
+)
+
+var (
+	k8s                        *openshift.Client
+	impersonatedResourceClient *openshift.Client
+	prom                       *prometheus.Client
+	clusterVersion             *apiv1.ClusterVersion
+	err                        error
+	upgradeConfig              upgradev1alpha1.UpgradeConfig
+)
+
+var _ = ginkgo.Describe("managed-upgrade-operator", ginkgo.Ordered, func() {
+	ginkgo.BeforeAll(func(ctx context.Context) {
+		log.SetLogger(ginkgo.GinkgoLogr)
+		var err error
+		k8s, err = openshift.New(ginkgo.GinkgoLogr)
+		Expect(err).ShouldNot(HaveOccurred(), "Unable to setup k8s client")
+		Expect(upgradev1alpha1.AddToScheme(k8s.GetScheme())).Should(BeNil(), "Unable to register upgradev1alpha1 api scheme")
```

Comment:
```suggestion
		Expect(upgradev1alpha1.AddToScheme(k8s.GetScheme())).Should(Succeed(), "Unable to register upgradev1alpha1 api scheme")
```

to make the assertion a bit more clear, we can be specific when we use `Expect(thing()).Should(BeNil())` vs something like `Expect(thing()).Should(Succeed())`

this goes for a lot of the assertions below as well